### ISSUE TITLE
404-page issue on first startup

### ISSUE THREAD

**Issue by enkelmedia** [umbraco/Umbraco-CMS#11589]
### Which *exact* Umbraco version are you using? For example: 9.0.1 - don't just write v9

9.1.0-rc

### Bug summary

Just tested the new 9.1.0-rc, version and the new 404 page does not work initially. I started up my site, configured a doctype, created content etc and published. Surfed the new node which worked but got an ugly error when trying the 404-page.

### Steps to reproduce

Installed templates,

```
dotnet new -i Umbraco.Templates::9.1.0-rc
```

```
dotnet new umbraco --name MyTestSite --SqlCe
```

Started the project, added a "Startpage"-doc type allowed at root and created a node in the content-tree. Surfing this page works. Then going to a URL that does not exist shows this:

![image](https://user-images.githubusercontent.com/1782524/140525627-3ca8161a-496b-4f0e-871f-94bd3b1a2576.png)

I'm running the site in hosting debug:true on VS2022 (Preview). Note: After a site restart this works so it might not be a big issue.

---
_This item has been added to our backlog [AB#15194](https://dev.azure.com/umbraco/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/15194)_

**Comment by bergmania** [comment]
Thanks for reporting.. 
This looks weird.. It looks like the IIS error page. Did you try this without IIS?

**Comment by enkelmedia** [comment]
@bergmania Hmm, I just started the site from Visual Studio so it was not running via IIS.

**Comment by Eyescream** [comment]
How did you try the 404?  
Because the url in the screenshot is something like account/accessdenied  .. that seems to have redirected to itself multiple times, until the querystring was too long to handle

**Comment by enkelmedia** [comment]
I started the Site from Visual Studio, this issue was only present right after thr installation. So adding Umbraco-packages, running the site, going trough installer, then trying 404.

After a restart it works as excepted.

**Comment by Zeegaan** [comment]
Hi there! Thanks for the issue report 🐛 
I had some trouble with this one, but basically this happens because when you have to install/upgrade umbraco we register the install endpoint, and now if you go to a page that doesn't exist, it tries to redirect you to the install endpoint (which you no longer have access to, because you've already installed/upgraded) 
But this should be fixed in: https://github.com/umbraco/Umbraco-CMS/pull/11710

**Comment by nikolajlauridsen** [comment]
This has been fixed and is heading into 9.1.2 🎉
